### DISCUSSION TITLE
[Logger]Is there a way to output log level with CRITICAL ?

### DISCUSSION THREAD

**Discussion by H1Gdev** [aws-powertools/powertools-lambda-typescript#1392] [Q&A]
Is there a way to output `CRITICAL` log like in `AWS Lambda Powertools for Python` ?

**Comment by dreamorosi** [comment]
Hi @H1Gdev, no, at the moment we don't support this level.

Powertools for Python builds on top of Python's [`logging` module](https://docs.python.org/3/library/logging.html) that is part of the standard library. As such, that version of Powertools supports the [same levels](https://docs.python.org/3/library/logging.html#logging-levels) that the `logging` module supports:

| Level | Numeric Value |
|--------|--------|
| `CRITICAL` | 50 |
| `ERROR` | 40 |
| `WARNING` | 30 |
| `INFO` | 20 |
| `DEBUG` | 10 | 
| `NOTSET` | 0 | 

Node.js on the other hand, deals with logs differently and doesn't have a strict level structure like Python does. The closest thing to Python's `logging` library in Node.js is the [`Console` object](https://nodejs.org/api/console.html#class-console).

The `Console` object does have some methods like `info`, `debug`, `warn`, `error`, etc. but doesn't have a concept of levels, **and most importantly for this discussion**, doesn't have a method that maps to `CRITICAL`. In Node.js, using one method or the other only influences whether the log will be emitted to `stdout` or `stderr` with `warn` and `error` going to `stderr` and all others going to `stdout`.

In Powertools for Typescript however the log level doesn't just influence the [which `Console` method will be used](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L648-L650) to log, but it also influences the fact that logs [will contain a `level` key](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L499) ([see docs](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#standard-structured-keys)).

With this in mind, would you mind sharing a bit more about your use case to understand why a `CRITICAL` level is needed and what would be the advantage over using the existent `ERROR` one?

**Reply by H1Gdev** [reply]
Thanks for your answer.

I would like to monitor `lambda function` logs by `level key`. (using `Amazon CloudWatch`)
In another `lambda function`(using Python), `ERROR` and `CRITICAL` are already used properly, so I would like to use them differently in new `lambda function`(using Node.js) under the same conditions.

I understand that there is no suitable method in Node.js.
Is there a way to set `level key` to `CRITICAL` in Node.js ?

**Reply by dreamorosi** [reply]
> Is there a way to set level key to CRITICAL in Node.js ?

At the moment you can't, and even if you tell TypeScript to ignore the type (or use JavaScript), it will log a warning and default to `INFO`.

> I would like to monitor lambda function logs by level key. (using Amazon CloudWatch)
In another lambda function(using Python), ERROR and CRITICAL are already used properly, so I would like to use them differently in new lambda function(using Node.js) under the same conditions.

Makes sense. Would you mind please [opening a feature request](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/new?assignees=&labels=type%2Ffeature-request%2Ctriage&template=feature_request.yml&title=Feature+request%3A+support%20%60CRITICAL%60%20log%20level) so that we can track interest and discuss implementing it?

**Comment by H1Gdev** [comment]
@dreamorosi 

Thank you for adding `CRITICAL` level.
